### PR TITLE
Add Checksum to Image Display

### PR DESCRIPTION
**Pull request by mwarning**
Show the hash/checksum beneath the image download link when the mouse hovers of the download link.

### CODE REVIEW

**Comment by mwarning** [comment]
@aparcar what do you think? Should we use "Checksum:", "Hash:" or even "SHA256:"?

**Comment by aparcar** [comment]
Hi, can you please add a screenshot? Also it would be great if you could propose this to the OpenWrt repository [here](https://gitlab.com/openwrt/web/firmware-selector-openwrt-org)

**Comment by aparcar** [comment]
I'd call it `sha256sum` which is the same as on downloads.openwrt.org. It both describes the used Algorithm and also is language independent, so we don't need to add it to the translations as all.

File: www/index.html [hunk 1]
```diff
@@ -103,6 +103,10 @@ <h3 id="downloads-title" class="tr-downloads">Download an image</h3>
 
           <!-- Explanatory text for each download button: make visible when hovering -->
           <div>
+            <div>
+              <span class="column tr-hash">Hash:</span>
+              <span id="image-hash"></span>
```
Comment (aparcar) [inline www/index.html:108]:
```suggestion
              <span id="image-sha256"></span>
```

File: www/index.js [hunk 2]
```diff
@@ -433,16 +440,27 @@ function updateImages(version, code, date, model, url, mobj, is_custom) {
...
-    $("#image-target").innerText = target;
-    $("#image-version").innerText = version;
-    $("#image-code").innerText = mobj["code"] || code;
-    $("#image-date").innerText = date;
+    setValue("#image-model", model);
+    setValue("#image-target", target);
+    setValue("#image-version", version);
+    setValue("#image-code", mobj["code"] || code);
+    setValue("#image-date", date);
+    setValue("#image-hash", undefined);
```
Comment (aparcar) [inline www/index.js:448]:
```suggestion
    setValue("#image-sha256", undefined);
```

File: www/index.js [hunk 3]
```diff
@@ -433,16 +440,27 @@ function updateImages(version, code, date, model, url, mobj, is_custom) {
...
 
     images.sort((a, b) => a.name.localeCompare(b.name));
 
-    for (const i in images) {
-      addLink(images[i].type, images[i].name);
+    for (const image of images) {
+      const a = createLink(image);
+
+      a.onmouseover = function () {
+        setValue("#image-hash", image.hash);
```
Comment (aparcar) [inline www/index.js:456]:
```suggestion
        setValue("#image-sha256", image.sha256);
```

**Review by aparcar — CHANGES_REQUESTED** [review]

**Comment by aparcar** [comment]
This is not a big thing but I'd rather keep the checksum variables descriptive, imagine we want to use sha3  or whatever else at some point...

File: www/index.html [hunk 4]
```diff
@@ -103,6 +103,10 @@ <h3 id="downloads-title" class="tr-downloads">Download an image</h3>
 
           <!-- Explanatory text for each download button: make visible when hovering -->
           <div>
+            <div>
+              <span class="column tr-hash">Hash:</span>
```
[the remark this answers is not in the document]
Comment (mwarning) [reply www/index.html:107]: We need to keep `class="column"` to apply CSS formatting.

[the remark this answers is not in the document]
Comment (aparcar) [reply www/index.html:107]: Oh oops!

**Comment by mwarning** [comment]
"sha256sum:" as label is a good idea.

**Comment by aparcar** [comment]
Thanks for the rework. Could you please create a PR [here](https://gitlab.com/openwrt/web/firmware-selector-openwrt-org/-/merge_requests) and to keep this repository in sync?

**Comment by mwarning** [comment]
@aparcar I haven't merged this because I am not quite happy about this:
- the design does look a bit odd, maybe it is just me
- the hash blows up the overview.png without being able to remove easily them

file

**Comment by aparcar** [comment]
What about reducing the font size of the hash? (Like in the patch i send you last week)

**Comment by mwarning** [comment]
Looks better that way:
file

**Comment by mwarning** [comment]
done

**Pull request closed without merging**